src/main/java/logic/unzip/UnzipFileToDirectoryController.java

```diff
@@ -0,0 +1,61 @@
+package logic.unzip;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class UnzipFileToDirectoryController implements UnzipFileToDirectoryable{
+    @Override
+    public void unzip(Path source, Path target) {
+        try {
+            ZipInputStream zis = new ZipInputStream(new FileInputStream(source.toFile()));
+            ZipEntry zipEntry = zis.getNextEntry();
+
+            while (zipEntry != null) {
+                boolean isDirectory = isDirectory(zipEntry);
+                Path newPath = zipSlipProtect(zipEntry, target);
+                copyFiles(isDirectory, newPath, zis);
+                zipEntry = zis.getNextEntry();
+
+            }
+            zis.closeEntry();
+        } catch (IOException ioException) {
+            System.out.println(ioException);
+        }
+    }
+
+    private boolean isDirectory(ZipEntry zipEntry) {
+        if (zipEntry.getName().endsWith(File.separator)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void copyFiles(Boolean isDirectory, Path newPath, ZipInputStream zis) throws IOException {
+        if (isDirectory) {
+            Files.createDirectories(newPath);
+        } else {
+            if (newPath.getParent() != null) {
```

Comment:
całkowicie bez sensu można to połaczyć w jedno i zrobić else if()

Reply:
Zrobione.